### PR TITLE
[FEATURE] reduced Filter

### PR DESCRIPTION
**Pull request by myapos**
Filter an array of objects based on condition and return array with reduced objects.

## Description

This function utility is using ES6 Array.reduce() in order to filter an array with objects based on condition.
If condition is met, then Array.reduce is used again to construct object with the predefined properties. Further more, in order to dynamically evaluate condition based in object properties, item must be passed through window to Function constructor. Finally, when new object is constructed then it is saved in the output array. 

## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: snippets/reducedFilter.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        return aggr;
+      }, {});
+      acc.push(parsedObj); 
+    } 
+    return (acc);
+  }, []);
+```
+#### Usage Example available in :
+
+`https://codepen.io/myapos/pen/dJGByW?editors=0112`
```
Comment (Chalarangelo) [inline snippets/reducedFilter.md]: Examples have to be inlined in the code, absolutely no external links.

File: snippets/reducedFilter.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+* Filtered array with new objects. Properties of new objects are a subset of
+properties of original objects 
+
+#### Info
+
+Used ES6 reduce 
+Dummy data for testing 
+Generated with http://www.mockaroo.com/ 
+
```
Comment (Chalarangelo) [inline snippets/reducedFilter.md]: Explain more thoroughly the techniques and methods used.

**Comment by myapos** [comment]
Ok I will be back with the necessary updates. Thanks for your review.

**Comment by iamsoorena** [comment]
beside the things my friends said. just FYI @myapos It's a very good practice to have more descriptive commit messages. :)

File: snippets/reducedFilter.md (hunk 2 above)
Comment (myapos) [reply snippets/reducedFilter.md]: Hello @Chalarangelo. I added more details in the update files. Could you please review it? Thanks.

File: snippets/reducedFilter.md (hunk 1 above)
Comment (myapos) [reply snippets/reducedFilter.md]: Added examples in comments

**Comment by myapos** [comment]
@Chalarangelo could you please more specific?

**Comment by atomiks** [comment]
What you're looking for is a callback function instead of an eval'd condition.

```js
const data = [{
  id: 1,
  name: 'john',
  age: 24
}, {
  id: 2,
  name: 'mike',
  age: 50
}];

const reducedFilter = (data, keys, callback) => 
  data.reduce((acc, item) => 
    callback(item)
      ? acc.concat(keys.reduce((acc, key) => {
          acc[key] = item[key];
          return acc;
        }, {}))
      : acc,
  []);

reducedFilter(data, ['id', 'name'], item => item.name.startsWith('m'));
// [{ id: 1, name: 'mike' }]
```

**Comment by skatcat31** [comment]
@atomiks callback is semantically incorrect here since it isn't async and makes it a little confusing(still good code though)

Also why concat instead of push?

```js
const reducedFilter = (data, keys, test) =>
  data.reduce(
    (acc, item) =>
      test(item)
        ? acc.push(
            keys.reduce((acc, key) => {
              acc[key] = item[key];
              return acc;
            }, {})
          )
        : acc,
    []
  );
```

**Comment by atomiks** [comment]
`push` returns the index, not the `acc` array

Also I thought callback wasn't necessarily an async term, it's just when a certain event occurs. For example, you'd your function is called by the `.filter/map/.forEach` methods despite them being sync,  making it a "callback"

**Comment by skatcat31** [comment]
callbacks are a generic terms for a function to be called in response to a trigger. Predicates are callbacks used as a trigger and will always be coerced to Booleans.

example of predicate versus callbacks:
```js
const ifElse = (predicate, ifcb, elsecb) => (...a) => predicate(...a) ? ifcb(...a) : elsecb(...a)
[1,2,3].map(ifElse(x => !x%2, x => x * 2, x =>x)) // [1,4,3] 
```

Again purely a flavor thing, but stating test or predicate will make the code much clearer in the long run, especially for beginners since if they see test or predicate they know it should return a truthy or a falsy instead of an any

**Pull request merged**